dev_docs/07_visual_testing_guide.md

```diff
@@ -0,0 +1,93 @@
+
+## Implementation details
+
+  The visual testing mechanism uses Puppeteer(for interacting with the testing environment and the rendered components), Jest-Puppeteer(to provide all required configuration to run tests using Puppeteer.), and Percy(to take snapshots for comparing visual diffs) to ensure components render correctly under various conditions. The key components include:
+
+1. **Configuration Files**:
+   - [***visual.index.js***](../jest.conf/visual.index.js): Configures Jest-Puppeteer and includes server checks to ensure the visual testing playground is up and running.
+   - [***visual.setup.js***](../jest.conf/visual.setup.js): Sets up global functions and constants needed for visual tests.
+
+2. **Utility Functions** [***(visual.testUtils.js)***](../jest.conf/visual.testUtils.js):
+   - `renderComponent(component, props, slots)`: Renders the specified component with given props and slots in the visual testing playground.
+   - `takeSnapshot(name, options)`: Takes a Percy snapshot with the given name and options.
+   - `click(selector)`, `hover(selector)`, `scrollToPos(selector, scrollOptions)`, `waitFor(selector)`, `delay(time)`: Utility functions for simulating user interactions.
+
+3. **Visual Testing Playground** ([***testing-playground.vue***](../docs/pages/testing-playground.vue)):
+   - A dedicated page for component visual testing, ensuring expected visual behavior under various conditions.
+   - Dynamically renders components based on messages received from the test runner.
+
+## Prerequisites
+
+- Make sure that your Node version is > 18.x with all the rquirements(listed in package.json) installed.
+- Ensure that you declare the `PERCY_TOKEN` environment varibale when runnning visual tests locally.
+- You should not have anything running on `port:4000` as the visual testing server utilizes that port to run the tests.
+
+## Guide to use visual testing mechanism
+
+### Running visual tests locally
+
+1. **Setup**:
+   - Ensure all dependencies are installed:
+
+     ```bash
+     yarn install
+     ```
+
+   - Set the `PERCY_TOKEN` environment variable:
+
+     ```bash
+     export PERCY_TOKEN=<your-percy-token>
+     ```
+
+2. **Run Visual Tests**:
+   - Execute the tests with the visual testing configuration:
+
+     ```bash
+     yarn test:visual
+     ```
+
+3. **Check results**:
+   - Head over to the Percy Dashboard and check the results for the latest Percy build.
+
+### Visual testing workflow
+
+  The visual testing workflow is integrated with GitHub Actions. When changes are made, a deployment request is initiated. Only Learning Equality team members can approve this request. Once approved, the deployment is executed, and visual tests are run automatically. The results of the test run are surfacedin the form of an automated comment containing a link to the Percy build.
```

Comment:
We could rewrite this to something like "We use GitHub Actions to execute the visual testing workflow on every PR. When changes to a PR are made, ..."

Reply:
We have also a typo here: `surfacedin` -> `surfaced in`